HTMLSerializer.js

```diff
@@ -458,6 +553,46 @@ var HTMLSerializer = class {
   }
 
   /**
+   * Asynchronously fill in any holes in |this.html|.
+   *
+   * @param {Document} doc The Document being serialized.
+   * @param {Function} callback The callback function, which will be called when
+   *     all asynchronous processing is finished.
+   */
+  fillHolesAsync(doc, callback) {
+    this.fillFontHoles(doc, callback);
+  }
+
+  /**
+   * Takes all of the cross origin stylesheets, processes their font
+   * declarations, and adds them to |this.html|. When finished, calls
+   * |this.fillSrcHoles|.
+   *
+   * @param {Document} doc The Document being serialized.
+   * @param {Function} callback The callback function, which will be called when
+   *     all asynchronous processing is finished.
+   */
+  fillFontHoles(doc, callback) {
+    if (this.crossOriginStyleSheets.length == 0) {
+      var fonts = `<style>${this.fontCSS.join('')}</style>`;
+      this.html[this.fontPlaceHolderIndex] = fonts;
+      this.fillSrcHoles(callback);
```

Comment:
Instead of this, pass a callback to fillFontHoles written like:

var me = this;
function() {
  me.fillSrcHoles(callback);
}